patches/tModLoader/Terraria.ModLoader/ModTexturedType.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+
+namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// The base type for most modded things with textures.
+	/// </summary>
+	public abstract class ModTexturedType:IAutoloadable
```

Comment:
Why does this not extend ModType ?

Reply:
It would lead to some weird method wrapping. It is very doable, but then it would have 2 AddInstance methods, one of which wouldn't do anything. If we wanted to add a base Texture property which would be assigned some time that might help... Currently the Name property doesn't autoassign either unless the AddInstance method calls a method which does it.

Actually, now with the major rework we will have for 1.4, maybe it would be a perfect time to just change the system entirely. Instead of autoload methods, what if we had autoload properties and made the ref name and texture fields just rely on overriding Name and Texture properties. If someone wants to have multiple instances of the same class with different name/textures, then they can override them with auto properties and set them in the constructor.

Reply:
That's not a bad idea, but what are the usecases (can we come up with a good example?)

Reply:
It is definitely more of an organizational change. In terms of functionality, it does allow for a single `ModType` class rather than two, like mentioned above.
In terms of organization though, it would help diversify the information into more meaningful locations. From a conceptual stand point, having to override Autoload in order to change the internal name or the texture doesn't make much sense. Also, if modders want to manually add a class, they wouldn't need to pass in the name or texture. Instead, they can keep this information within the class it relates to.

Reply:
`having to override Autoload in order to change the internal name or the texture doesn't make much sense`
Yeah that's a good statement. I'm definitely in favor of the change, will ask others.